flare/ase/otf.py

```diff
@@ -325,15 +362,28 @@ def as_dict(self):
 
         for key in ["output", "pred_func", "structure", "dft_input", "md"]:
             dct.pop(key)
+        dct["md"] = self.md.todict()
 
         return dct
 
     @staticmethod
     def from_dict(dct):
-        flare_calc = FLARE_Calculator.from_file(dct["flare_calc"])
+        flare_calc_dict = json.load(open(dct["flare_calc"]))
+        if flare_calc_dict["class"] == "FLARE_Calculator":
+            flare_calc = FLARE_Calculator.from_file(dct["flare_calc"])
+            _kernels = None
+        elif flare_calc_dict["class"] == "SGP_Calculator":
+            from flare_pp.sparse_gp_calculator import SGP_Calculator
+
+            flare_calc, _kernels = SGP_Calculator.from_file(dct["flare_calc"])
+        else:
+            raise TypeError(
+                f"The calculator from {dct['flare_calc']} is not recognized."
+            )
```

Comment:
These changes are a bit hard to follow -- a comment would be helpful

Reply:
the kernel needs to be returned, especially when we build SGP. It is still the issue we had before that if the kernel is not returned to the current space, the variable of it (like `kernel->hyperparameters`) can not be found and raises segmentation fault.

comments are added in commit f3c0b55